javaparser-core/src/main/java/com/github/javaparser/ast/validator/chunks/ModifierValidator.java

```diff
@@ -49,7 +49,7 @@ public void visit(ClassOrInterfaceDeclaration n, ProblemReporter reporter) {
 
     private void validateClassModifiers(ClassOrInterfaceDeclaration n, ProblemReporter reporter) {
         if (n.isTopLevelType()) {
-            validateModifiers(n, reporter, PUBLIC, ABSTRACT, FINAL, STRICTFP);
+            validateModifiers(n, reporter, PUBLIC, ABSTRACT, STATIC, FINAL, STRICTFP);
```

Comment:
Here is an example of this in the Checker Framework.
https://github.com/Bohdankm22/checker-framework/blob/master/checker/jtreg/nullness/issue824/First.astub
Inner interfaces are written at the top level. Same way static nested classes could be declared.
Here are two ways of the issues resolution:
- Handle static classes/interfaces at the top-level (already implemented)
- After parsing a stub file look for the inner/nested classes/interfaces at the top-level and try to move it inside outer classes.